src/ui/App.tsx

```diff
@@ -55,16 +63,39 @@ function App(): JSX.Element {
 
   const [loadingThreads, setLoadingThreads] = useState(LoadState.initial);
 
+  async function* fetchThreadsWithMetadata(): AsyncGenerator<Thread> {
+    // TODO: Sort by date.
+    const threads = (await fetchThreads(`in:inbox -in:${LabelName.keep}`))
+      .threads;
+
+    for (const rawThread of defined(threads)) {
+      const thread = new Thread(rawThread);
+      await thread.fetchMessages();
+      if (!thread.hasMessagesInInbox()) {
+        await modifyThread(thread.id, [], ['INBOX']);
```

Comment:
In this case I think we want to continue to the next thread rather than yielding?